### PR TITLE
[TAS-3968] 💄 Hide Crisp widget on page

### PR DESCRIPTION
**Pull request by AuroraHuang22**
https://github.com/user-attachments/assets/b5d623f8-94b7-44af-a921-9f06bd2906d6

### CODE REVIEW

File: pages/account.vue [hunk 1]
```diff
@@ -211,21 +198,24 @@ const { handleError } = useErrorHandler()
...
 onMounted(() => {
-  listenCrispLoaded(() => {
+  const { onLoaded } = useScriptCrisp({ id: crispId })
+
+  onLoaded(() => {
     isCrispLoaded.value = true
+    window.$crisp?.push(['do', 'chat:hide'])
+
+    window.$crisp?.push(['on', 'chat:closed', () => {
+      window.$crisp?.push(['do', 'chat:hide'])
```
Comment (williamchong) [inline pages/account.vue]: Maybe keep the icon if it is opened once, not sure

File: pages/account.vue [hunk 2]
```diff
@@ -211,21 +198,24 @@ const { handleError } = useErrorHandler()
...
-const { instance: crisp, onLoaded: listenCrispLoaded } = useScriptCrisp({ id: crispId.value })
+const isCrispLoaded = ref(false)
 
 onMounted(() => {
-  listenCrispLoaded(() => {
+  const { onLoaded } = useScriptCrisp({ id: crispId })
+
+  onLoaded(() => {
     isCrispLoaded.value = true
+    window.$crisp?.push(['do', 'chat:hide'])
```
Comment (williamchong) [inline pages/account.vue]: Can we get the crisp instance from `useScriptCrisp` instead?

File: pages/account.vue [hunk 3]
```diff
@@ -261,14 +251,14 @@ async function handleLikerPlusButtonClick() {
...
-
 async function handleCustomerServiceLinkButtonClick() {
-  useLogEvent('customer_service', { method: 'link' })
+  if (!isCrispLoaded.value) {
+    window.open(crispChatURL.value, '_blank')
+    useLogEvent('customer_service', { method: 'link' })
+
+    return
+  }
+  window.$crisp?.push(['do', 'chat:show'])
```
Comment (williamchong) [inline pages/account.vue]: Can we get the crisp instance from `useScriptCrisp` instead?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: pages/account.vue [hunk 4]
```diff
@@ -211,21 +198,19 @@ const { handleError } = useErrorHandler()
...
+  url.searchParams.set('website_id', crispId)
   return url.toString()
 })
-
-const { instance: crisp, onLoaded: listenCrispLoaded } = useScriptCrisp({ id: crispId.value })
+const isCrispLoaded = ref(false)
+const { instance: crisp, onLoaded } = useScriptCrisp({ id: crispId })
 
 onMounted(() => {
-  listenCrispLoaded(() => {
```
Comment (williamchong) [inline pages/account.vue:227]: please keep this name unchanged if no special reason

File: pages/account.vue [hunk 5]
```diff
@@ -261,14 +246,15 @@ async function handleLikerPlusButtonClick() {
...
-  if (crisp && isCrispLoaded.value) {
-    crisp.do('chat:open')
+async function handleCustomerServiceLinkButtonClick() {
+  if (!isCrispLoaded.value) {
+    window.open(crispChatURL.value, '_blank')
+    useLogEvent('customer_service', { method: 'link' })
+
+    return
   }
-}
```
Comment (williamchong) [inline pages/account.vue:245]: Should we log an event here?

File: pages/account.vue [hunk 6]
```diff
@@ -211,21 +198,19 @@ const { handleError } = useErrorHandler()
 useHead({
   title: $t('account_page_title'),
 })
-
-const isCrispLoaded = ref(false)
-// XXX: `id` in `useScriptCrisp()` is broken, it always returns `'crisp'` even `id` is provided and it is also not reactive 💩
-const crispId = computed(() => config.public.scripts.crisp.id)
+const crispId = String(config.public.crispWebsiteId || '5c009125-5863-4059-ba65-43f177ca33f7')
```
Comment (williamchong) [inline pages/account.vue]: What would be the type of this if we dont call String() ?

**Review by williamchong — APPROVED** [review]

File: pages/account.vue (hunk 6 above)
Comment (nwingt) [reply pages/account.vue]: are `config.public.scripts.crisp.id` & `crispWebsiteId` the same thing?

Comment (williamchong) [reply pages/account.vue]: probably yes

File: pages/account.vue [hunk 7]
```diff
@@ -261,14 +246,15 @@ async function handleLikerPlusButtonClick() {
   }
 }
 
-async function handleCustomerServiceChatButtonClick() {
-  useLogEvent('customer_service', { method: 'chat' })
-  if (crisp && isCrispLoaded.value) {
-    crisp.do('chat:open')
+async function handleCustomerServiceLinkButtonClick() {
+  if (!isCrispLoaded.value) {
```
Comment (nwingt) [inline pages/account.vue]: does `@click` event trigger even `:disabled="!isCrispLoaded"`?

File: pages/account.vue (hunk 6 above)
Comment (nwingt) [reply pages/account.vue]: @AuroraHuang22  please use `config.public.scripts.crisp.id` instead, maybe set `5c009125-5863-4059-ba65-43f177ca33f7` in `nuxt.config.ts` too

File: pages/account.vue [hunk 8]
```diff
@@ -181,6 +156,18 @@
...
+            <UButton
+              :label="$t('account_page_contact_support')"
+              variant="outline"
+              leading-icon="i-material-symbols-contact-support"
+              color="neutral"
+              :disabled="!isCrispLoaded"
+              size="lg"
+              @click="handleCustomerServiceLinkButtonClick"
+            />
+          </div>
```
Comment (nwingt) [inline pages/account.vue]: Why changing the button style?

Current
Screenshot 2025-06-19 at 01 21 23

Change
Screenshot 2025-06-19 at 01 22 07

File: nuxt.config.ts [hunk 9]
```diff
@@ -56,6 +56,7 @@ export default defineNuxtConfig({
       },
       airtableCMSBaseId: process.env.AIRTABLE_CMS_BASE_ID,
       airtableCMSProductsTableId: process.env.AIRTABLE_CMS_PRODUCTS_TABLE_ID,
+      crispWebsiteId: process.env.NUXT_PUBLIC_SCRIPTS_CRISP_ID,
```
[the remark this answers is not in the document]
Comment (nwingt) [reply nuxt.config.ts]: Invalid

File: pages/account.vue (hunk 7 above)
Comment (nwingt) [reply pages/account.vue]: @AuroraHuang22 button is hidden when Crisp is not loaded
Screenshot 2025-06-19 at 09 36 57

**Review by nwingt — CHANGES_REQUESTED** [review]

File: composables/use-logger.ts [hunk 10]
```diff
@@ -50,8 +50,12 @@ export function useLogEvent(eventName: string, eventParams: EventParams = {}) {
     console.error(`Failed to track event with Meta Pixel: ${eventName}`, eventParams)
   }
 
-  const { instance: crisp } = useScriptCrisp()
+  const config = useRuntimeConfig()
+  const crispId = String(config.public.scripts.crisp.id)
+
+  const { instance: crisp } = useScriptCrisp({ id: crispId })
   if (crisp) {
+    crisp.do('chat:hide')
```
Comment (williamchong) [inline composables/use-logger.ts]: Move `const { onLoaded: listenCrispLoaded } = useScriptCrisp()` to a plugin to do this globally instead?

File: pages/account.vue [hunk 11]
```diff
@@ -211,21 +199,19 @@ const { handleError } = useErrorHandler()
...
+const isCrispLoaded = ref(false)
+const { instance: crisp, onLoaded } = useScriptCrisp({ id: crispId })
 
 onMounted(() => {
-  listenCrispLoaded(() => {
+  onLoaded(() => {
     isCrispLoaded.value = true
+    crisp?.do('chat:hide')
   })
 })
```
Comment (williamchong) [inline pages/account.vue]: Move `const { onLoaded: listenCrispLoaded } = useScriptCrisp()` to a plugin to do this globally instead?

**Review by williamchong — CHANGES_REQUESTED** [review]

**Review by williamchong — APPROVED** [review]
Please test on testnet to see if any edge case behaviour doesn't work

File: pages/account.vue [hunk 12]
```diff
@@ -211,23 +198,13 @@ const { handleError } = useErrorHandler()
...
-
-const isCrispLoaded = ref(false)
-// XXX: `id` in `useScriptCrisp()` is broken, it always returns `'crisp'` even `id` is provided and it is also not reactive 💩
-const crispId = computed(() => config.public.scripts.crisp.id)
+const crispId = String(config.public.scripts.crisp.id)
 const crispChatURL = computed(() => {
   const url = new URL('https://go.crisp.chat/chat/embed')
-  url.searchParams.set('website_id', crispId.value || '5c009125-5863-4059-ba65-43f177ca33f7')
+  url.searchParams.set('website_id', (crispId || '5c009125-5863-4059-ba65-43f177ca33f7'))
```
Comment (nwingt) [inline pages/account.vue]: Unnecessary brackets 
```suggestion
  url.searchParams.set('website_id', crispId || '5c009125-5863-4059-ba65-43f177ca33f7')
```

File: pages/account.vue [hunk 13]
```diff
@@ -211,23 +198,13 @@ const { handleError } = useErrorHandler()
 useHead({
   title: $t('account_page_title'),
 })
-
-const isCrispLoaded = ref(false)
-// XXX: `id` in `useScriptCrisp()` is broken, it always returns `'crisp'` even `id` is provided and it is also not reactive 💩
-const crispId = computed(() => config.public.scripts.crisp.id)
+const crispId = String(config.public.scripts.crisp.id)
```
Comment (nwingt) [inline pages/account.vue]: I think no need to cast this to string

Comment (nwingt) [inline pages/account.vue]: I think no need to cast this to string

Comment (nwingt) [reply pages/account.vue]: If the typing is missing, try `npm run dev` again

Comment (nwingt) [reply pages/account.vue]: If the typing is missing, try `npm run dev` again

File: pages/account.vue [hunk 14]
```diff
@@ -261,14 +238,15 @@ async function handleLikerPlusButtonClick() {
   }
 }
 
-async function handleCustomerServiceChatButtonClick() {
-  useLogEvent('customer_service', { method: 'chat' })
-  if (crisp && isCrispLoaded.value) {
-    crisp.do('chat:open')
+async function handleCustomerServiceLinkButtonClick() {
+  if (!crisp) {
+    window.open(crispChatURL.value, '_blank')
```
Comment (nwingt) [inline pages/account.vue]:
```suggestion
    await navigateTo(crispChatURL.value, {
      external: true,
      open: { target: '_blank' },
    })
```

File: plugins/crisp.ts [hunk 15]
```diff
@@ -0,0 +1,10 @@
+export default defineNuxtPlugin(() => {
+  const config = useRuntimeConfig()
+  const crispId = String(config.public.crispWebsiteId)
```
Comment (nwingt) [inline plugins/crisp.ts]: Please use config.scripts.crisp.id without String casting

**Review by nwingt — CHANGES_REQUESTED** [review]
Actually, can we inject `crispChatURL` inside the plugin

**Review by nwingt — APPROVED** [review]
